Yoast/Yoast-SEO-for-TYPO3
# [TASK] Moved the snippet preview call (for CMS9) and save scores call to backend context, refactoring and small cleanup

## Summary

This PR can be summarized in the following changelog entry:

* Moved the snippet preview ajax call to the backend (only for CMS9, CMS8 still needs the frontend for URL generation)
* Moved the save scores ajax call to the backend for both CMS8 and CMS9
* Refactored some classes, bit of cleanup and function-splitups
* Moved the preview typoscript to CMS8.typoscript
* Deleted the yoast_seo_save_scores typoscript and Preview.typoscript file
* Introduced a new PreviewService which both the AjaxController and SnippetPreview userFunction use
* Moved standard labels to LocaleService
* Added the `TYPO3_SITE_PATH` constant to the `getUrlForType` function to fix the problem with installations on a subdomain

## Relevant technical choices:

* Prominent words url has not been changed because that requires a change in the premium package

## Test instructions

This PR can be tested by following these steps:

* Pull this branch within a CMS9 and CMS8 installation and test if the snippet preview still works (for pages and TCA records)
* See if this fixes the problem with an installation on subdomain, this is now only applicable to CMS8 since CMS9 does this through backend

## Quality assurance
